mycore-media/src/main/java/org/mycore/media/services/MCRImageThumbnailGenerator.java

```diff
@@ -0,0 +1,71 @@
+package org.mycore.media.services;
+
+import org.jdom2.JDOMException;
+import org.mycore.datamodel.niofs.MCRPath;
+import org.mycore.imagetiler.MCRImage;
+import org.mycore.imagetiler.MCRTiledPictureProps;
+import org.mycore.iview2.services.MCRIView2Tools;
+
+import javax.imageio.ImageReader;
+import java.awt.Graphics2D;
+import java.awt.RenderingHints;
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+import java.nio.file.FileSystem;
+import java.nio.file.Path;
+import java.util.regex.Pattern;
+
+public class MCRImageThumbnailGenerator implements MCRThumbnailGenerator {
+
+    private String matchingMimetype = "^image\\/.*";
+
+    @Override
+    public boolean matchesFileType(String mimeType, MCRPath path) {
+        Pattern p = Pattern.compile(matchingMimetype);
+        return p.matcher(mimeType).matches();
+    }
+
+    @Override
+    public BufferedImage getThumbnail(MCRPath path, int size) throws IOException {
+        BufferedImage level1Image = getImage(path);
```

Comment:
getImage() has no information on the size parameter. You should read the next bigger image and scale it down